akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -540,7 +535,9 @@ private[akka] final case class Log[T](name: String, extract: T ⇒ Any, logAdapt
     logLevels = ctx.attributes.logLevels.getOrElse(DefaultLogLevels)
     log = logAdapter match {
       case Some(l) ⇒ l
-      case _       ⇒ Logging(ctx.materializer.asInstanceOf[ActorFlowMaterializer].system, DefaultLoggerName)
+      case _ ⇒
+        val mat = ActorFlowMaterializer.downcast(ctx.materializer) // TODO if not in Actor materializer we could fallback to println?
```

Comment:
can't we throw a more context specific exception if we cannot downcast? Like logger is only usable with the actor based materializer?